src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+        return key == str;
+    }
+
+private:
+    const char* _str = nullptr;
+};
+template <typename T, PropertyMutability M>
+inline std::ostream& operator<<(std::ostream& os, const BaseKey<T, M>& key) {
+    return os << key.str();
+}
+}  // namespace util
+/** @endcond */
+
+/**
+ * @brief This class us used to bind property key string with paramter type
+ * @tparam T type of parameter used to pass or get property
+ */
+template <typename T, PropertyMutability mutability_ = PropertyMutability::RW>
+struct Key : public util::BaseKey<T, mutability_> {
+    using util::BaseKey<T, mutability_>::BaseKey;
+    /**
+     * @brief Constructs property
+     * @tparam Args property constructor arguments types
+     * @param args property constructor arguments
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Args>
+    inline std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {this->str(), Any::make<T>(std::forward<Args>(args)...)};
+    }
+};
+
+/**
+ * @brief This class us used to bind property key string with read only paramter type
+ * @tparam T type of parameter used to pass or get property
+ */
+template <typename T>
+struct Key<T, PropertyMutability::RO> : public util::BaseKey<T, PropertyMutability::RO> {
+    using util::BaseKey<T, PropertyMutability::RO>::BaseKey;
+};
+
+/**
+ * @brief Read only property to get a std::vector<PropertyName> of supported read only properies.
+ *
+ * This can be used as an compiled model property as well.
+ *
+ */
+static constexpr Key<std::vector<PropertyName>, PropertyMutability::RO> supported_properties{"SUPPORTED_PROPERTIES"};
+
+/**
+ * @brief Read only property to get a std::vector<std::string> of available device IDs
+ */
+static constexpr Key<std::vector<std::string>, PropertyMutability::RO> available_devices{"AVAILABLE_DEVICES"};
+
+/**
+ * @brief Read only property to get a std::vector<std::string> of optimization options per device.
+ */
+static constexpr Key<std::vector<std::string>, PropertyMutability::RO> optimization_capabilities{
+    "OPTIMIZATION_CAPABILITIES"};
+
+/**
+ * @brief Read only property which defines support of import/export functionality by plugin
+ */
+static constexpr Key<bool, PropertyMutability::RO> import_export_support{"IMPORT_EXPORT_SUPPORT"};
+
+/**
+ * @brief Read only property to get a name of model_name
+ */
+static constexpr Key<std::string, PropertyMutability::RO> model_name{"NETWORK_NAME"};
+
+/**
+ * @brief Read only property to get an unsigned integer value of optimal number of compiled model infer requests.
+ */
+static constexpr Key<uint32_t, PropertyMutability::RO> optimal_number_of_infer_requests{
+    "OPTIMAL_NUMBER_OF_INFER_REQUESTS"};
+
+namespace hint {
+
+/**
+ * @brief Enum to define possible performance measure hints
+ */
+enum class ModelPriority {
+    LOW = 0,
+    MEDIUM = 1,
+    HIGH = 3,
+};
```

Comment:
is it GPU only?
CC @vladimir-paramuzov I suppose for GPU we have host / queue / device priorities. Is this property valid for GPU?